samples/snippets/csharp/VS_Snippets_CLR/conceptual.isolatedstorage/cs/source.cs

```diff
@@ -5,21 +5,52 @@
 
 public class CreatingFilesDirectories
 {
-    public static void Main()
+public static void Main()
+{
+    using (IsolatedStorageFile isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Domain | IsolatedStorageScope.Assembly, null, null))
     {
-        using (IsolatedStorageFile isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Domain | IsolatedStorageScope.Assembly, null, null))
+        isoStore.CreateDirectory("TopLevelDirectory");
+        isoStore.CreateDirectory("TopLevelDirectory/SecondLevel");
+        var isofilename = "AnotherTopLevelDirectory/NewFile.txt";
+        try
         {
-            isoStore.CreateDirectory("TopLevelDirectory");
-            isoStore.CreateDirectory("TopLevelDirectory/SecondLevel");
-            isoStore.CreateDirectory("AnotherTopLevelDirectory/InsideDirectory");
-            Console.WriteLine("Created directories.");
+            isoStore.CreateFile(isofilename);
+        }
+        catch (IsolatedStorageException iex){
+            Console.WriteLine(iex.ToString());
+        }
+
+        var path = Path.GetDirectoryName(isofilename);
+        isoStore.CreateDirectory(path);//Creating "AnotherTopLevelDirectory";
+        isoStore.CreateFile(isofilename);//Creating "AnotherTopLevelDirectory/NewFile.txt";
+        
+        isoStore.CreateDirectory("AnotherTopLevelDirectory/InsideDirectory");
+        Console.WriteLine("Created directories.");
 
-            isoStore.CreateFile("InTheRoot.txt");
-            Console.WriteLine("Created a new file in the root.");
+        isoStore.CreateFile("InTheRoot.txt");
+        Console.WriteLine("Created a new file in the root.");
 
-            isoStore.CreateFile("AnotherTopLevelDirectory/InsideDirectory/HereIAm.txt");
-            Console.WriteLine("Created a new file in the InsideDirectory.");
+        isoStore.CreateFile("AnotherTopLevelDirectory/InsideDirectory/HereIAm.txt");
+        Console.WriteLine("Created a new file in the InsideDirectory.");
         }
     }
+
+    private string CreateFolders(string folderName, string fileName)
```

Comment:
I'm curious why you added this method @OzBob It's not referenced anywhere.

Reply:
removed private function - it was there in my original draft as a more robust code example, but is not necessary for this code snippet

spacing set to 4 spaces.